Incorrect import in the multitargetting folder for MSBuild

**Bicep version**
0.4.1008

**Describe the bug**
In `bicep/src/Bicep.MSBuild/buildMultiTargeting/` there's an incorrect import.

**To Reproduce**
Steps to reproduce the behavior:

```
.nuget\packages\azure.bicep.msbuild\0.4.1008\buildMultiTargeting\Azure.Bicep.MSBuild.props(2,3): error MSB4019: The imported project ".nuget\packages\azure.bicep.msbuild\0.4.1008\build\BicepTasks.props" was not found. Confirm that the expression in the Import declaration "..\build\BicepTasks.props" is correct, and that the file exists on disk.
```

**Additional context**
A fix could be similar to `<Import Project="..\build\$(MSBuildThisFileName).$(RuntimeIdentifier).props" />` as well as there's a need to have a e2e nuget est with `TargetFrameworks` to exercise the multitargetting folder.

--- a/src/Bicep.MSBuild.E2eTests/examples/empty/empty.proj
+++ b/src/Bicep.MSBuild.E2eTests/examples/empty/empty.proj
@@ -3,9 +3,7 @@
 -->
 <Project Sdk="Microsoft.Build.NoTargets">
   <PropertyGroup>
-    <!-- Ensure the package supports multi-targeting -->
-    <TargetFrameworks>net5;net472</TargetFrameworks>
-    <BicepOutputPath>bin\$(Configuration)\templates</BicepOutputPath>
+    <TargetFramework>net472</TargetFramework>
   </PropertyGroup>
 
   <!-- 
@@ -17,10 +15,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Bicep Include="passthrough.bicep"/>
-    <Bicep Include="empty.bicep"/>
-    <Bicep Include="subdir\theAnswer.bicep">
-      <OutputFile>$(BicepOutputPath)\special\special.arm</OutputFile>
-    </Bicep>
+    <!-- Do not set any Bicep item: this ensures the package correctly does not error on projects without Bicep items -->
   </ItemGroup>
 </Project>

--- a/src/Bicep.MSBuild.E2eTests/examples/missingCli/missingCli.proj
+++ b/src/Bicep.MSBuild.E2eTests/examples/missingCli/missingCli.proj
@@ -16,7 +16,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <!-- the following file name is intentionally wrong -->
     <Bicep Include="empty.bicep"/>
   </ItemGroup>
 </Project>

--- a/src/Bicep.MSBuild.E2eTests/examples/multiple/multiple.proj
+++ b/src/Bicep.MSBuild.E2eTests/examples/multiple/multiple.proj
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <!-- Ensure the package supports multi-targeting -->
     <TargetFrameworks>net5;net472</TargetFrameworks>
+    <LanguageTargets>$(MSBuildThisFileDirectory)\multitarget-mitigation.targets</LanguageTargets> <!-- https://github.com/microsoft/MSBuildSdks/issues/155 -->
     <BicepOutputPath>bin\$(Configuration)\templates</BicepOutputPath>
   </PropertyGroup>
 

--- a/src/Bicep.MSBuild.E2eTests/examples/multiple/multitarget-mitigation.targets
+++ b/src/Bicep.MSBuild.E2eTests/examples/multiple/multitarget-mitigation.targets
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+    <!-- Mitigation for https://github.com/microsoft/MSBuildSdks/issues/155 -->
+    <Import Project="$(MSBuildToolsPath)\Microsoft.Common.CrossTargeting.targets" Condition="$(IsCrossTargetingBuild) == 'true'" />
+    <Import Project="$(MSBuildToolsPath)\Microsoft.Common.targets" Condition="$(IsCrossTargetingBuild) != 'true'" />
+</Project>

--- a/src/Bicep.MSBuild/Bicep.MSBuild.csproj
+++ b/src/Bicep.MSBuild/Bicep.MSBuild.csproj
@@ -18,7 +18,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <!-- do not upgrade these packages - we need to support mesbuild 16 -->
+    <!-- do not upgrade these packages - we need to support msbuild 16 -->
     <PackageReference Include="Microsoft.Build.Framework" Version="16.10.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="16.10.0" PrivateAssets="All" />
   </ItemGroup>

--- a/src/Bicep.MSBuild/build/Azure.Bicep.MSBuild.targets
+++ b/src/Bicep.MSBuild/build/Azure.Bicep.MSBuild.targets
@@ -7,7 +7,7 @@
     <BicepOutputPath Condition=" $(BicepOutputPath) == '' ">$(OutputPath)</BicepOutputPath>
   </PropertyGroup>
 
-  <Target Name="BicepCompile" Condition=" @(Bicep) != '' " DependsOnTargets="$(BicepCompileDependsOn)" AfterTargets="$(BicepCompileAfterTargets)" BeforeTargets="$(BicepCompileBeforeTargets)">
+  <Target Name="BeforeBicepCompile" Condition=" @(Bicep) != '' " BeforeTargets="BicepCompile">
     <ItemGroup>
       <_BicepResolved Include="@(Bicep)">
         <!-- populate missing output file metadata using the default path -->
@@ -16,13 +16,13 @@
 
       <_BicepOutputFile Include="%(_BicepResolved.OutputFile)" />
     </ItemGroup>
-
     <!-- pre-create directories for all the outputs in case they don't exist -->
     <MakeDir Directories=" %(_BicepOutputFile.RootDir)%(Directory)" />
-
     <Error Text="The path to the Bicep compiler is not set. Reference the appropriate Azure.Bicep.CommandLine.* package for your runtime or set the BicepPath property." Condition=" $(BicepPath) == '' " />
-    <Bicep SourceFile="%(_BicepResolved.FullPath)" OutputFile="%(OutputFile)" ToolExe="$(BicepPath)" />
-    
+  </Target>
+
+  <Target Name="BicepCompile" Inputs="@(_BicepResolved)" Outputs="%(_BicepResolved.OutputFile)" DependsOnTargets="$(BicepCompileDependsOn)" AfterTargets="$(BicepCompileAfterTargets)" BeforeTargets="$(BicepCompileBeforeTargets)">
+    <Bicep SourceFile="%(_BicepResolved.FullPath)" OutputFile="%(OutputFile)" ToolExe="$(BicepPath)" YieldDuringToolExecution="true" />
     <Message Importance="High" Text="$(MSBuildProjectName) -&gt; %(_BicepResolved.OutputFile)" />
 
     <ItemGroup>

--- a/src/Bicep.MSBuild/buildMultiTargeting/Azure.Bicep.MSBuild.props
+++ b/src/Bicep.MSBuild/buildMultiTargeting/Azure.Bicep.MSBuild.props
@@ -1,3 +1,3 @@
 <Project>
-  <Import Project="..\build\BicepTasks.props" />
+  <Import Project="..\build\Azure.Bicep.MSBuild.props" />
 </Project>

--- a/src/Bicep.MSBuild/buildMultiTargeting/Azure.Bicep.MSBuild.targets
+++ b/src/Bicep.MSBuild/buildMultiTargeting/Azure.Bicep.MSBuild.targets
@@ -1,3 +1,3 @@
 <Project>
-  <Import Project="..\build\BicepTasks.targets" />
+  <Import Project="..\build\Azure.Bicep.MSBuild.targets" />
 </Project>

